File: flask-backend/models/order_item.py
from extensions import db


class OrderItem(db.Model):
    __tablename__ = "order_items"

    orderItemId = db.Column(db.Integer, primary_key=True)
    orderId = db.Column(db.Integer, db.ForeignKey("orders.orderId"), nullable=False)
    productId = db.Column(
        db.Integer, db.ForeignKey("products.productId"), nullable=False
    )
    quantity = db.Column(db.Integer, default=1)
    price = db.Column(db.Float, nullable=False)

    product = db.relationship("Product", lazy="joined")

File: spring-boot-backend/src/main/java/com/springbootbackend/model/OrderItem.java
package com.springbootbackend.model;
import java.util.List;

import com.springbootbackend.model.Order;
import com.springbootbackend.model.Product;
import jakarta.persistence.*;
import lombok.*;

@Data
@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "order_Items")
public class OrderItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long orderItemId;

    @ManyToOne
    @JoinColumn(name="order_id",nullable = false)
    private Order order;

    @ManyToOne
    @JoinColumn(name = "product_id",nullable = false)
    private Product product;

    private Integer quantity;
    private Double price;
}
